### PR TITLE
Feature/calendar heatmap

### PR DESCRIPTION
**Pull request by dalbanhi**
## New Features Implemented
- [x] Implemented calendar heat map for contributions using dummy data with some measure of responsiveness
heatmap-calendar

### Styling
- [x] Removed vertical centering from main layout
``` javascript
// (main)/layout.tsx
<main className="flex min-h-screen w-full justify-between xl:w-4/5">
  <LeftSidebar />
  {children}
  <RightSidebar />
</main>
```
- [x] Used text-transform instead of `capitalizeFirstLetter` function:
``` javascript
// components/shared/FilterPill.tsx
<span
        className={` text-caption capitalize hover:text-myWhite-100 ${textColor} `}
>
```
- [x] Used text-overflow `truncate` Tailwind class to deal with overflow text
``` javascript
 <span
        className={`truncate text-p3Med ${textColor} hover:text-myWhite-100`}
>
```

### Structure
- [x] Used consistent exporting of constant variables in constants file for using dummy data (see `constants/index.ts`)

### CODE REVIEW

File: components/shared/Home/ContributionsTracker.tsx [hunk 1]
```diff
@@ -1,133 +1,86 @@
...
-            return (
-              <div key={index} className="text-p4Med text-myWhite-300">
-                {day}
-              </div>
-            );
-          })}
+        showWeekdayLabels={true}
+        gutterSize={4}
+        weekdayLabels={DAY_LABELS}
+        classForValue={(value) => getClassNameValue(value)}
```
Comment (brandonetter) [inline components/shared/Home/ContributionsTracker.tsx]:
```suggestion
        classForValue={getClassNameValue}
```

Minor suggestion- we don't need to make an anonymous function here.

File: src/app/(main)/explore/page.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  searchParams: { [key: string]: string | string[] | undefined };
+}) => {
+  const posts = await getPosts(searchParams.filter as PostType);
+  console.log(posts);
+  return (
+    <section>
+      <RecentPostsHeader />
+      {posts.map((post, index) => {
+        return <PostCard key={index} post={post} isShort={false} />;
+      })}
```
Comment (brandonetter) [inline src/app/(main)/explore/page.tsx]:
```suggestion
      {posts.map((post) => {
        return <PostCard key={post.id} post={post} isShort={false} />;
      })}
```

Eventually when we have several filters- we'll likely start returning some results multiple times depending on the search. By using a key that is unique to the item we're mapping over, we're letting the react renderer know that this is the same item. If we just use an index, the react virtual dom won't be able to keep track of the components properly.

https://blog.anja-stricker.de/why-using-index-as-key-in-reactjs-map-function-can-cause-problems-and-how-to-fix-them

https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66


It likely won't be an issue when just displaying results, but if we have the ability to use a unique identified like an ID, let's default to using it to make our code as robust as possible.

File: src/app/globals.css [hunk 3]
```diff
@@ -23,3 +30,46 @@ body {
...
+  fill: #55597d;
+}
+
+.react-calendar-heatmap .react-calendar-heatmap-weekday-label {
+  padding-right: 5px;
+}
+
+.react-calendar-heatmap-weekday-labels {
+  transform: translate(0px, 14px);
+}
```
Comment (brandonetter) [inline src/app/globals.css:75]: https://www.loom.com/share/82e91e675dde4bd5b1b61e59e71a8e48

Looks great already- don't think you need to change anything. But this is what we were talking about before with attempting to allow the 'overflow' on the svg itself.

Your solution may be best- as this feature technically doesn't have wide availability as a standard. I've only tested on three browsers- it's possible certain desktop or mobile browsers would just clip the text!

https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/overflow

File: components/shared/Home/ContributionsTracker.tsx [hunk 4]
```diff
@@ -1,133 +1,86 @@
...
+        </p>
+        <div className="flex items-center justify-center gap-2">
+          <span className="text-subtitle text-myWhite-300">Less</span>
+          <svg className="h-[15px] w-[54px] max-md:h-[5px] max-md:w-[24px] ">
+            <g>
+              {rectClasses.map((rectClass, index) => {
+                return (
+                  <rect
+                    key={index}
+                    x={
```
Comment (brandonetter) [inline components/shared/Home/ContributionsTracker.tsx:69]: key as index is fine here as we're mapping over a static, unchanging list of items.

**Review by brandonetter — CHANGES_REQUESTED** [review]
Good work. Minor changes regarding the anon function and just some notes to continue our previous discussion.

Couple changes and we'll merge 👍

File: components/shared/Home/ContributionsTracker.tsx [hunk 5]
```diff
@@ -1,133 +1,86 @@
...
   return (
-    <div className=" flex flex-col border border-myWhite-300 bg-myBlack-900 p-4">
-      <div className="mt-4 flex justify-around gap-2">
-        {months.map((month, index) => {
-          return (
-            <div key={index} className="text-p4Med text-myWhite-300">
-              {month}
-            </div>
-          );
+    <div className=" rounded-sm border border-myWhite-500 bg-myBlack-900 p-4">
```
Comment (brandonetter) [inline components/shared/Home/ContributionsTracker.tsx]: I think the correct border color in the design is:
`border-myBlack-700`

**Review by brandonetter — CHANGES_REQUESTED** [review]

**Comment by dalbanhi** [comment]
Hey @brandonetter  I implemented the small changes you mentioned. I'm not 100% sure how to reference them, but they should be available (I see that what you mentioned is 'outdated'). Let me know if there's anything else! Thank you!

**Pull request merged**